### ISSUE TITLE
UTM Tracking in the URL

### ISSUE THREAD

**Issue by rhyswynne** [ibericode/koko-analytics#193]
Switching from GA4 to Koko.

It'd be really handy if it could track utm campaign variables in URLs. So if I send an email out promoting a black friday sale to new newsletter subscribers, a potential URL could be:-

www.example.com/product?utm_source=new+subscribers&utm_medium=email&utm_campaign=black+friday+sale&utm_content=first+cta+button

That way I know how much traffic comes from that segment.

These are standard in GA, so I'd really like Koko to interpret these.

**Comment by rhyswynne** [comment]
Just seen this - https://github.com/ibericode/koko-analytics/issues/118 - hopefully that issue gets added with a "Feature Suggestion" label. As it'd be great to push.

**Comment by dannyvankooten** [comment]
Hey @rhyswynne,

Good to see you on here and thank you for chiming in! I agree, UTM would be really neat. I'll have to give it a proper thinking to determine in what form this would warrant its own database structure and everything that comes with it, but for now I am at least happy to share that this will be possible in some form with [event tracking](https://www.kokoanalytics.com/kb/tracking-events/).

Note that this is currently a feature of Koko Anaytics Pro, which is currently in closed beta. It will replace [our OpenCollective initiatieve](https://opencollective.com/koko-analytics) in order to hopefully raise some funds and secure future deveopment and cover support costs for the free plugin.

What it allows is for you to register custom events (eg one caled "UTM Campaign") and then allow you to track events of this type grouped by an optional parameter. For example, calling `koko_analytics.trackEvent('UTM Campaign', 'My Christmas Campaign')` would increment the count for `My Christmas Campaign` with 1.

You can see the versatility of this feature in action at the bottom of the [tracking events](https://www.kokoanalytics.com/kb/tracking-events/) page. The "was this helpful" section is using a custom event with either a positive or a negative value to help keep track of negatively rated KB articles. You can see what this looks like on the dashboard in our [live demo here](https://www.kokoanalytics.com/?go-to-demo=1).

I'll be writing a ton more docs on this in the next  few weeks and will include UTM tracking as an example.

TLDR: I'd love to support UTM variables in some form at some point, but at least in the near future custom events will be a step in that direction!

**Comment by rhyswynne** [comment]
Thanks Danny! Yeah that would be a game changer for me and I think the approach you've shared above would be great. I'm assuming the trackEvent would be able to read URL parameters? :)
